skyportal/handlers/api/source.py

```diff
@@ -2810,37 +2823,40 @@ def post(self):
         if data.get("sourceId") is None:
             return self.error("Missing required parameter `sourceId`")
 
-        source = Obj.get_if_accessible_by(data["sourceId"], self.current_user)
-        if source is None:
-            return self.error('Source not found', status=404)
-
-        source_id = data["sourceId"]
+        with self.Session() as session:
 
-        source_group_ids = [
-            row[0]
-            for row in Source.query_records_accessible_by(
-                self.current_user, columns=[Source.group_id]
-            )
-            .filter(Source.obj_id == source_id)
-            .all()
-        ]
+            source = session.scalars(
+                Obj.select(session.user_or_token).where(Obj.id == data["sourceId"])
+            ).first()
+            if source is None:
+                return self.error('Source not found', status=404)
+
+            source_id = data["sourceId"]
+
+            source_group_ids = [
+                row[0]
+                for row in session.scalars(
+                    Source.scalars(
+                        session.user_or_token, columns=[Source.group_id]
+                    ).where(Source.obj_id == source_id)
+                ).all()
+            ]
 
-        if bool(set(group_ids).difference(set(source_group_ids))):
-            forbidden_groups = list(set(group_ids) - set(source_group_ids))
-            return self.error(
-                "Insufficient recipient group access permissions. Not a member of "
-                f"group IDs: {forbidden_groups}."
-            )
+            if bool(set(group_ids).difference(set(source_group_ids))):
+                forbidden_groups = list(set(group_ids) - set(source_group_ids))
```

Comment:
I'm wondering if this is giving the user too much information. In other cases we just say "you don't have enough access to some groups" and don't specify which ones.